src/cdk/scrolling/virtual-scroll-viewport.ts

```diff
@@ -243,8 +247,12 @@ export class CdkVirtualScrollViewport implements OnInit, OnDestroy {
    * (in pixels).
    */
   setRenderedContentOffset(offset: number, to: 'to-start' | 'to-end' = 'to-start') {
+    // For a horizontal viewport in a right-to-left language we need to translate along the x-axis
+    // in the negative direction.
+    const axisDirection = this.orientation == 'horizontal' && this._dir && this._dir.value == 'rtl'
+        ? -1 : 1;
```

Comment:
Since you check them a couple of time, I would add `_isHorizontal` and `_isRtl` methods. Then this would just look like
```ts
// For a horizontal viewport in a right-to-left language we need to translate along the x-axis
// in the negative direction.
const axisCoefficient = this._isHorizontal && this._isRtl() ? -1 : 1;
```